client/asset/dcr/spv.go

```diff
@@ -765,33 +766,39 @@ func (w *spvWallet) UnlockAccount(ctx context.Context, pw []byte, _ string) erro
 // SyncStatus returns the wallet's sync status.
 // Part of the Wallet interface.
 func (w *spvWallet) SyncStatus(ctx context.Context) (bool, float32, error) {
-	bestHeight := w.bestPeerInitialHeight()
-	_, h := w.dcrWallet.MainChainTip(ctx)
-	height := h
-	if h > bestHeight {
-		bestHeight = h
+	targetHeight := w.bestPeerInitialHeight()
+	if targetHeight == 0 {
+		return false, 0, nil
 	}
 
-	if h == 0 || bestHeight == 0 {
+	_, height := w.dcrWallet.MainChainTip(ctx)
+	if height == 0 {
 		return false, 0, nil
 	}
 
-	synced, progress := w.spv.Synced(), float32(height)/float32(bestHeight)
+	if height > targetHeight {
+		targetHeight = height
+	}
+
+	synced, progress := w.spv.Synced(), float32(height)/float32(targetHeight)
 	if progress > 0.999 && !synced {
 		progress = 0.999
 	}
 
 	return synced, progress, nil
 }
 
-// bestPeerInitialHeight is the highest InitialHeight recorded from our peers.
+// bestPeerInitialHeight is the highest InitialHeight recorded from connected
+// spv peers. This value is maintained even if the peer that reported the value
+// later becomes disconnected.
```

Comment:
I wonder if this is an attack vector.  Is the spv syncer smart enough not to include peers with bogus chain heights?

Reply:
Probably not something we should rely on. Another thought I had, which now sounds more reasonable, is to only use the cached value if there are no more connected peers; otherwise the value reported by connected peers takes precedence. Will update.